qos/evm/check.go

```diff
@@ -50,19 +60,53 @@ func getEndpointCheck(
 }
 
 // withChainIDCheck updates the request context to make an EVM JSON-RPC eth_chainId request.
+// eg. '{"jsonrpc":"2.0","id":1,"method":"eth_chainId"}'
 func withChainIDCheck(requestCtx *requestContext) {
 	requestCtx.jsonrpcReq = buildJSONRPCReq(idChainIDCheck, methodChainID)
 }
 
 // withBlockHeightCheck updates the request context to make an EVM JSON-RPC eth_blockNumber request.
+// eg. '{"jsonrpc":"2.0","id":1,"method":"eth_blockNumber"}'
 func withBlockHeightCheck(requestCtx *requestContext) {
 	requestCtx.jsonrpcReq = buildJSONRPCReq(idBlockNumberCheck, methodBlockNumber)
 }
 
-func buildJSONRPCReq(id int, method jsonrpc.Method) jsonrpc.Request {
-	return jsonrpc.Request{
+// withArchivalBlockCheck updates the request context to make an EVM JSON-RPC eth_getBalance request with a random archival block number.
+// eg. '{"jsonrpc":"2.0","id":1,"method":"eth_getBalance","params":["0x28C6c06298d514Db089934071355E5743bf21d60", "0xe71e1d"]}'
+func withArchivalBlockCheck(requestCtx *requestContext) {
+	// Get the archival block number from the endpoint store.
+	archivalCheckConfig := requestCtx.endpointStore.serviceState.archivalCheckConfig
+	// Get the current state of the archival check.
+	serviceArchivalState := requestCtx.endpointStore.serviceState.archivalState
+
+	requestCtx.jsonrpcReq = buildJSONRPCReq(
+		idArchivalBlockCheck,
+		methodGetBalance,
+		// Pass params in this order, eg. "params":["0x28C6c06298d514Db089934071355E5743bf21d60", "0xe71e1d"]
+		// Reference: https://ethereum.org/en/developers/docs/apis/json-rpc/#eth_getbalance
+		archivalCheckConfig.ContractAddress,
+		serviceArchivalState.blockNumberHex,
+	)
+
+	// Set the archival balance check flag to true.
+	// This is used to ensure that only hydrator requests for the archival block number are used
```

Comment:
I read this a couple of times, but it's still unclear to me why we need it.

Can we go without it? If not, can you try to update the comment so its clearer?

Reply:
No longer needed. Has been removed.